### PR TITLE
Install singularity in the integration GitHub workflow

### PR DESCRIPTION
**Pull request by nsoranzo**
so that test classes that need it are not skipped any more.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by bernt-matthias** [comment]
`test/integration/test_containerized_jobs.py::TestSingularityJobsIntegration::test_build_mulled` should be fixed by https://github.com/galaxyproject/galaxy/pull/15657

**Review by bernt-matthias — COMMENTED** [review]
Should we set `apptainer-version`?

**Comment by bernt-matthias** [comment]
Then maybe like so https://github.com/nsoranzo/galaxy/pull/12 :)

**Comment by nsoranzo** [comment]
> Should we set `apptainer-version`?

I'd rather keep it unpinned unless something breaks.

**Comment by nsoranzo** [comment]
This is ready to review at last, thanks @bernt-matthias !

**Comment by bernt-matthias** [comment]
> This is ready to review at last, thanks @bernt-matthias !

The PR had a very good timing. Needed to understand those tests better anyway for https://github.com/galaxyproject/galaxy/pull/15614 :)

**Review by mvdbeek — APPROVED** [review]
Thanks @nsoranzo & @bernt-matthias!

File: test/integration/embedded_pulsar_singularity_job_conf.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ execution:
       runner: pulsar_embed
       singularity_enabled: true
       singularity_required: true
+      # Since tests run in /tmp/ , we apparently need to forbid the default mounting of /tmp
+      singularity_run_extra_arguments: '--no-mount tmp'
```
Comment (mvdbeek) [inline test/integration/embedded_pulsar_singularity_job_conf.yml:17]: Completely bonkers defaults btw. Who said something about singularity being safer than docker 🤣

Comment (mvdbeek) [reply test/integration/embedded_pulsar_singularity_job_conf.yml:17]: Should we make that the default ?

Comment (bernt-matthias) [reply test/integration/embedded_pulsar_singularity_job_conf.yml:17]: Good question? Was actually my first idea .. but I had the feeling that I do not understood the actual problem well enough. Just for our reference: The actual problem in the tests was: 

```
FATAL:   container creation failed: mount /tmp/tmpfr6yrqr6/000/1->/tmp/tmpfr6yrqr6/000/1 error: while mounting /tmp/tmpfr6yrqr6/000/1: destination /tmp/tmpfr6yrqr6/000/1 doesn't exist in container
```

The problem seems only to occur when mounting subdirs of `/tmp`, i.e. `-B /tmp/:/tmp/` works and `-B /tmp/tmpei26zwmg/:/tmp/tmpei26zwmg/ ` doesn't.

Comment (mvdbeek) [reply test/integration/embedded_pulsar_singularity_job_conf.yml:17]: It's totally unsafe to mount /tmp by default, docker doesn't do it.

Comment (bernt-matthias) [reply test/integration/embedded_pulsar_singularity_job_conf.yml:17]: Created an issue for now https://github.com/galaxyproject/galaxy/issues/15673

**Pull request merged**